libcontainer/cgroups/v1_utils.go

```diff
@@ -200,19 +181,18 @@ func getCgroupMountsHelper(ss map[string]bool, mi io.Reader, all bool) ([]Mount,
 		if len(m.Subsystems) > 0 || all {
 			res = append(res, m)
 		}
-	}
-	if err := scanner.Err(); err != nil {
-		return nil, err
+		if numFound >= len(ss) {
+			break
+		}
```

Comment:
Is this new check necessary? It's also not correct if `all` is `true`.

Reply:
I haven't changed the original logic here, which was
```go
for scanner.Scan() && numFound < len(ss) {
```

but it's good that you asked, as now I see that this function is not tested with `all` set to true.

Reply:
Ah, OK, the `all` logic was added by https://github.com/opencontainers/runc/pull/1049/commits/f55799640143fea4d6ca699ddc2aac72e3e56d7a (done for https://github.com/google/cadvisor/pull/1476) and it correctly avoided incrementing `numFound` in case `all == true`....

.... until https://github.com/opencontainers/runc/pull/1817/commits/5ee0648bfbe037bcf4b36b2d53e59cc91bf42b55 (PR #1817) broke it (since v1.0.0-rc6 and until now).

I am not sure why e.g. google/cadvisor#2309 (that bumps runc to a version with this bug) haven't reintroduced the https://github.com/google/cadvisor/issues/1461. There is _still_ an open issue that looks very similar to it though: https://github.com/google/cadvisor/issues/1843.

Anyway, I am fixing this in https://github.com/opencontainers/runc/pull/2689,
and rebasing this PR on top of that one.